ML-model/Task1-Data_scraping/web_scraper.py

```
@@ -1,30 +1,127 @@
+            # Find all article elements using Selenium
+            articles = driver.find_elements(By.CSS_SELECTOR, "xpl-results-item")
+
+            print(f"Found {len(articles)} articles on page {page}")
+
+            for article in articles:
+                try:
+                    # Extract title using XPath
+                    title_elem = None
+                    try:
+                        title_elem = article.find_element(By.XPATH, ".//h3/a")
+                    except:
```

Ruff (0.8.2)

Do not use bare `except`

(E722)